File: aoj/alds1/a.py
def main():
    n = int(input())
    target = list(map(int, input().split()))

    for i in range(len(target)):
        key = target[i]
        j = i - 1
        while j >= 0 and target[j] > key:
            target[j+1] = target[j]
            j -= 1
        target[j+1] = key
        print(*target)

if __name__ == "__main__":
    main()

File: aoj/alds1/a.kt
fun main() {
    val n = readLine()!!.toInt()
    val target = readLine()!!.split(" ").map { it.toInt() }.toMutableList()

    for (i in target.indices) {
        val key = target[i]
        var j = i - 1
        while (j >= 0 && target[j] > key) {
            target[j + 1] = target[j]
            j--
        }
        target[j + 1] = key
        println(target.joinToString(" "))
    }
}

